server/api/integrations/index.post.ts

```
@@ -1,44 +1,31 @@
 import { DB as db } from "../../sqlite-service";
-import { integrations as integrationsTable, mastodonIntegrationOptionSchema, mapIntegrationOptionSchema } from "../../../db/schema";
+import { integrations as integrationsTable, integrationOptionsSchema, IntegrationOptions } from "../../../db/schema";
 import { updateIntegrations } from "../../../shared/utils/abilities";
 import { z } from 'zod';
 
-const postMastodonBodySchema = z.object({
+const integrationBodySchema = z.object({
   enable: z.boolean({coerce: true}),
-  name: z.literal("mastodon"),
-  options: mastodonIntegrationOptionSchema,
+  options: integrationOptionsSchema,
 });
 
-const postMapBodySchema = z.object({
-  enable: z.boolean({coerce: true}),
-  name: z.literal("map"),
-  options: mapIntegrationOptionSchema,
-});
 
 export default defineEventHandler(async (event) => {
   // @ts-ignore TODO https://github.com/nuxt/nuxt/issues/29263
   await authorize(event, updateIntegrations);
 
-  const { name } = await readBody(event);
-
-  let integrationData;
-
-  if (name === 'mastodon') {
-    integrationData = await readValidatedBody(event, postMastodonBodySchema.parse);
-  }
-  else if (name === 'map') {
-    integrationData = await readValidatedBody(event, postMapBodySchema.parse);
-  }
-  else {
+  const maybeIntegrationBody = await readValidatedBody(event, integrationBodySchema.safeParse);
```

instead of calling `.safeParse` you can call `.parse` and let it throw if it fails validation. Nuxt will handle sending the correct http response. If you do it as I described it will give you detailed error data in dev and terse structured errors in prod. Then you can delete it if success check below. Example [here](https://github.com/unfaresf/unfaresf/blob/main/server/api/broadcasts/index.post.ts#L16). Can you change it?